attributes/alibaba.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+cred_dir = ::File.expand_path(::File.join('~', '.alibaba'))
+credential_file = ::File.join(cred_dir, 'credentials')
+
+TIMESTAMP = Time.now.strftime('%Y_%m_%d_%H_%M_%S').freeze
+INVALID_AMI_NAME_CHARACTERS = /[^\w\(\)\.\-\/]/
+
+def normalize_name(name)
+  # Append timestamp
+  # Remove invalid characters
+  # Ensure it's 128 characters or less
+  "#{name}_#{TIMESTAMP}".gsub(INVALID_AMI_NAME_CHARACTERS, '').strip[0..127]
+end
+
+default['marketplace_image']['alibaba']['cred_dir'] = cred_dir
+default['marketplace_image']['alibaba']['credential_file'] = credential_file
+
+default['marketplace_image']['alibaba']['public']['compliance']['enabled'] = false
+default['marketplace_image']['alibaba']['public']['automate']['enabled'] = false
+
+default['marketplace_image']['alibaba']['product_urls'] = {
+  'marketplace' => 'http://chef-software.oss-cn-beijing.aliyuncs.com/chef-marketplace_0.2.5%2B20170911103840.git.2.1c6a67f-1_amd64.deb',
```

Comment:
I suspect that there's a firewall preventing the instance from accessing our package mirrors? Do we want to require the full URL or just the version of packages? What's the process of uploading a new package version to the Alibaba mirror? That should at least be documented.

Reply:
It is not a preventing of getting to mirrors, it is throttling. It is so slow to download that most of the time things time out.

At the moment we need the full URL to the package that needs to be downloaded. It was the simplest way I could think to get it working.

Currently it is a case of logging into the Alibaba cloud with the account in LastPass and manually uploading. There are command line tools that allow you to this but I have not investigated thoe yet.